runtime/v1/linux/proc/init.go

```diff
@@ -113,6 +113,9 @@ func (p *Init) Create(ctx context.Context, r *CreateConfig) error {
 		pio     *processIO
 		pidFile = newPidFile(p.Bundle)
 	)
+	p.pid.Lock()
```

Comment:
Curious about this lock.
If the pid is 0, can the rest of the code check that instead of holding a lock during this whole function?

Reply:
this is the create function so its safe to hold the lock until it's done